### PR TITLE
RF cum_concat_step simplify and other RF things

### PR DESCRIPTION
**Pull request by albertz**
(This is just here to at least run the test once before I merge it into main.)

(The other commits accidentally went into here, but it cannot hurt. Let's just not squash them together but rebase then.)

### CODE REVIEW

File: returnn/frontend/array_.py [hunk 1]
```diff
@@ -507,13 +512,18 @@ def cum_concat_step(
...
     if not out_spatial_dim:
         out_spatial_dim = axis + 1
-    # noinspection PyProtectedMember
-    return (
-        source._raw_backend.cum_concat_step(source, prev_accum=prev_accum, axis=axis, out_spatial_dim=out_spatial_dim),
-        out_spatial_dim,
+    out, (out_spatial_dim,) = rf.pad(
+        prev_accum, axes=[axis], padding=[(0, 1)], out_dims=[out_spatial_dim], value=source, handle_dynamic_dims=True
     )
+    return out, out_spatial_dim
```
Comment (NeoLegends) [inline returnn/frontend/array_.py:526]: I approve the idea here. Seems like `rf.pad` does not like padding w/ entire tensors though. CI says `value` needs to be scalar.

**Comment by albertz** [comment]
> What about the other backends? It seems strange to me that only one backend can pad w/ non-scalar values. If we only implement this for torch I'd rather adjust the RF `cum_conat_step` impl to not rely on `rf.pad` but a more general operation that works across all backends, no?.

There is not really any other backend currently which fully works and is actively used, so I don't think this has such a high priority now.

**Comment by NeoLegends** [comment]
But is it much work to change the impl to be generic by default?

**Comment by albertz** [comment]
I don't exactly understand? Via this PR, `rf.cum_concat_step` becomes generic, so this PR here fixes this?

**Comment by NeoLegends** [comment]
I mean wrt. `rf.pad` of non-scalar values.

**Comment by albertz** [comment]
But I still don't understand? `rf.pad` never supported non-scalar values? Now it does for the PT backend.

**Pull request merged**